node_modules/gh-orgunit/lib/internal/dao.js

```diff
@@ -415,3 +416,377 @@ var getOrgUnitCalendar = module.exports.getOrgUnitCalendar = function(orgUnit, s
+        _allOrgUnitIds = _.union(_allOrgUnitIds, childOrgUnitIds);
+        _getChildOrgUnits(childOrgUnitIds, callback, _allOrgUnitIds);
+    });
+};
+
+/**
+ * Get the ids of the direct children of a set of organisational units
+ *
+ * @param  {Number[]}   parentOrgUnitIds            The ids of the parent organisational units
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error object, if any
+ * @param  {Number[]}   callback.allOrgUnitIds      The ids of the organisational units that are direct children of the given set of organisational unit ids
+ * @api private
+ */
+var _getDirectChildOrgUnits = function(parentOrgUnitIds, callback) {
+    var options = {
+        'attributes': ['id'],
+        'where': {
+            'ParentId': parentOrgUnitIds
+        }
+    };
+    DB.OrgUnit.findAll(options).complete(function(err, childOrgUnits) {
+        if (err) {
+            return callback(err);
+        }
+
+        var ids = _.map(childOrgUnits, function(childOrgUnit) {
+            return childOrgUnit.id;
+        });
+        return callback(null, ids);
+    });
+};
+
+/**
+ * Export the organisational units, event series and events under an organisational unit to CSV
+ *
+ * @param  {Number}         rootId              The id of the root organisational unit in the tree of organisational units, event series and events
+ * @param  {OrgUnit[]}      orgUnits            All the organisational units that are in the tree. Each organisational unit should have the event series and events available on its object that are under it
+ * @param  {Function}       callback            Standard callback function
+ * @param  {Object}         callback.err        An error object, if any
+ * @param  {String}         callback.data       The exported organisational units, event series and events as CSV
+ * @see gh-orgunit/lib/rest.js for more information
+ * @api private
+ */
+var _exportOrgUnitCSV = function(rootId, orgUnits, callback) {
+    // Get the simplified tree by exporting it to JSON first
+    _exportOrgUnitJSON(rootId, orgUnits, function(err, exportedOrgUnit) {
+        if (err) {
+            return callback(err);
+        }
+
+        var columns = _getExportCSVColumns(exportedOrgUnit);
+        if (!columns) {
+            return callback({'code': 400, 'msg': 'Tried to export an organisational unit that has children with the same type of different levels'});
+        }
+
+        var records = _getExportCSVRecords(exportedOrgUnit);
+
+        // Stringify the records into CSV
+        var options = {
+            'header': true,
+            'columns': columns
+        };
+        return csv.stringify(records, options, callback);
+    });
+};
+
+/**
+ * Get the column headers for a CSV export of an organisational unit tree. If the tree contained
+ * the same type of organisational unit on different levels, this function will return `null`.
```

Comment:
I don't fully understand `If the tree contained the same type of organisational unit on different levels`. Can you clarify with an example?